### PR TITLE
feat: WT-2152 Add wallet connect enable disable option

### PR DESCRIPTION
**Pull request by mimi-imtbl**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

https://immutable.atlassian.net/browse/WT-2152

# Summary

# Detail and impact of the change

## Security

# Anything else worth calling out?

### CODE REVIEW

File: packages/checkout/sdk/src/widgets/definitions/configurations/widget.ts [hunk 1]
```diff
@@ -8,6 +8,12 @@ export type WidgetConfiguration = {
   /** The theme of the Checkout Widget (default: "DARK") */
   theme?: WidgetTheme;
   language?: WidgetLanguage;
+  /** Enable/disable Wallet Connect */
+  walletConnect?: {
+    enable: boolean;
+    includeWalletIds: string[];
+    excludeWalletIds: string[];
```
Comment (ZacharyCouchman) [inline packages/checkout/sdk/src/widgets/definitions/configurations/widget.ts:15]: Hey @mimi-imtbl , sorry can we remove these two arrays for now until we enable wallet connect whitelisting for partners

**Review by sharonsheah — DISMISSED** [review]
🚀

File: packages/checkout/sdk/src/widgets/definitions/configurations/widget.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ export type WidgetConfiguration = {
   /** The theme of the Checkout Widget (default: "DARK") */
   theme?: WidgetTheme;
   language?: WidgetLanguage;
+  /** Enable/disable Wallet Connect */
+  walletConnect?: boolean;
```
Comment (ZacharyCouchman) [inline packages/checkout/sdk/src/widgets/definitions/configurations/widget.ts:12]:
```suggestion
  walletConnect?: {
     enable: boolean
  },
```

Comment (ZacharyCouchman) [reply packages/checkout/sdk/src/widgets/definitions/configurations/widget.ts:12]: Sorry @mimi-imtbl , this is what I meant by remove the arrays for now. We want to be able to add them in later without it being a breaking change to the config object

**Review by ZacharyCouchman — APPROVED** [review]

**Review by sharonsheah — DISMISSED** [review]

**Review by dreamoftrees — APPROVED** [review]
👍

**Review by ZacharyCouchman — APPROVED** [review]

**Review by sharonsheah — APPROVED** [review]

**Pull request merged**